### PR TITLE
Gaob/relperm

### PR DESCRIPTION
**Pull request by gaobhub**
I have all changes here, including:
(1) depression storage evaluator bug
(2) add omega kr
(3) revised air resistance, but set default coeffs to 0. So if do not define those values in input file, it is the same with the old air resistance.
(4) use Sellers' soil resistance instead of Sakagucki's.

### CODE REVIEW

File: src/pks/flow/constitutive_relations/wrm/rel_perm_frzBC_evaluator.hh [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/*
+  Copyright 2010-202x held jointly by participating institutions.
+  ATS is released under the three-clause BSD License.
+  The terms of use and "as is" disclaimer for this license are
+  provided in the top-level COPYRIGHT file.
+
+  Authors: Ethan Coon (ecoon@lanl.gov)
```
Comment (ecoon) [inline src/pks/flow/constitutive_relations/wrm/rel_perm_frzBC_evaluator.hh]: Ethan Coon is not the author...

File: src/pks/flow/constitutive_relations/wrm/rel_perm_frzBC_evaluator_reg.hh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+*/
+
+#include "rel_perm_frzBC_evaluator.hh"
+
+namespace Amanzi {
+namespace Flow {
+
+// registry of method
+Utils::RegisteredFactory<Evaluator, RelPermFrzBCEvaluator>
+  RelPermFrzBCEvaluator::factory_("FrzBC rel perm");
```
Comment (ecoon) [inline src/pks/flow/constitutive_relations/wrm/rel_perm_frzBC_evaluator_reg.hh:17]: Please come up with a better, more explicit name.  "FrzBC" doesn't mean anything to me.  Longer is better in this case!

Comment (ecoon) [reply src/pks/flow/constitutive_relations/wrm/rel_perm_frzBC_evaluator_reg.hh:17]: You can leave the file named what it is, just update the "evaluator type"!

File: src/pks/surface_balance/constitutive_relations/land_cover/seb_physics_funcs.cc [hunk 3]
```diff
@@ -280,9 +279,11 @@ UpdateEnergyBalanceWithSnow_Inner(const GroundProperties& surf,
 
   // latent heat
   double vapor_pressure_skin = SaturatedVaporPressure(snow.temp);
-
+  double u_star = met.Us * c_von_Karman / std::log(met.Z_Us / CalcRoughnessFactor(snow.height, surf.roughness, snow.roughness));
+  double Re0 = params.density_air * u_star * CalcRoughnessFactor(snow.height, surf.roughness, snow.roughness) / params.dynamic_viscosity_air;
+  double KB = params.Da0_a * std::pow(Re0, params.Da0_b) - (params.Cd0_c * std::log(Re0) + params.Cd0_d);
```
Comment (ecoon) [inline src/pks/surface_balance/constitutive_relations/land_cover/seb_physics_funcs.cc]: Can you update the documentation and/or comments in this file with a reference as to where these equations are coming from?

File: src/pks/surface_balance/constitutive_relations/land_cover/seb_physics_defs.hh [hunk 4]
```diff
@@ -44,22 +44,27 @@ static const double c_R_ideal_gas = 461.52;                 // [Pa m^3 kg^-1 K^-
...
+      H_vaporization(2497848.),      // Latent heat of vaporization ------ [J/kg]
+      Cp_air(1004.0),                // Specific heat of air @ const pres- [J/K kg]
+      Cv_water(4218.636),            // Specific heat of water ----------- [J/K kg]
+      P_atm(101325.),                // atmospheric pressure ------------- [Pa]
+      gravity(9.807),                // gravity [kg m / s^2]
+      evap_transition_width(100.),   // transition on evaporation from surface to
+                                     // evaporation from subsurface [m],
+                                     // THIS IS DEPRECATED
+      KB(0.),
+      Da0_a(0.),
```
Comment (ecoon) [inline src/pks/surface_balance/constitutive_relations/land_cover/seb_physics_defs.hh:64]: please document what these magic numbers are

File: src/pks/surface_balance/constitutive_relations/land_cover/seb_physics_defs.hh [hunk 5]
```diff
@@ -72,19 +77,25 @@ struct ModelParams {
       plist.get<double>("evaporation transition width [Pa]", evap_transition_width);
 
     KB = plist.get<double>("log ratio between z0m and z0h [-]", KB);
+    Da0_a = plist.get<double>("coefficient a for Da0 [-]", Da0_a);
+    Da0_b = plist.get<double>("exponent b for Da0 [-]", Da0_b);
+    Cd0_c = plist.get<double>("coefficient c for Cd0 [-]", Cd0_c);
+    Cd0_d = plist.get<double>("coefficient d for Cd0 [-]", Cd0_d);
```
Comment (ecoon) [inline src/pks/surface_balance/constitutive_relations/land_cover/seb_physics_defs.hh:83]: what are Da0 and Cd0?

**Review by ecoon — CHANGES_REQUESTED** [review]
A few minor tweaks and documentation requests.

File: src/pks/surface_balance/constitutive_relations/land_cover/seb_physics_funcs.cc (hunk 3 above)
Comment (gaobhub) [reply src/pks/surface_balance/constitutive_relations/land_cover/seb_physics_funcs.cc]: Can we just remove these changes to air resistance? I am still hesitating to add this to the master branch.

File: src/pks/flow/constitutive_relations/wrm/rel_perm_frzBC_evaluator.hh [hunk 6]
```diff
@@ -0,0 +1,119 @@
+/*
+  Copyright 2010-202x held jointly by participating institutions.
+  ATS is released under the three-clause BSD License.
+  The terms of use and "as is" disclaimer for this license are
+  provided in the top-level COPYRIGHT file.
+
+  Authors: Ethan Coon (ecoon@lanl.gov), Bo Gao (gaob@ornl.gov)
+*/
+
+//! Evaluates relative permeability using water retention models.
```
Comment (ecoon) [inline src/pks/flow/constitutive_relations/wrm/rel_perm_frzBC_evaluator.hh:10]: please update the documentation of this.  This documentation is copy-pasted from rel_perm_evaluator, which is a generic evaluator that uses WRMs to compute krel -- the WRM is the thing that evaluates Krel.  Here you evaluate Krel internally (using the omega model).  Please add:

1. a description (e.g empirical model based on ... that works in freezing conditions...)
2. a citation
3. make sure any parameters (omega, etc) are listed here as a part of the model spec.

**Review by ecoon — CHANGES_REQUESTED** [review]

**Pull request merged**